trackintel/preprocessing/staypoints.py

```diff
@@ -72,27 +84,34 @@ def generate_locations(
             db = DBSCAN(eps=epsilon, min_samples=num_samples, algorithm="ball_tree", metric=distance_metric)
 
         if agg_level == "user":
-            location_id_counter = 0
-            # TODO: change into groupby
-            for user_id_this in ret_stps["user_id"].unique():
-                # Slice staypoints array by user. This is not a copy!
-                user_staypoints = ret_stps[ret_stps["user_id"] == user_id_this]
 
-                if distance_metric == "haversine":
-                    # the input is converted to list of (lat, lon) tuples in radians unit
-                    p = np.array([[radians(g.y), radians(g.x)] for g in user_staypoints.geometry])
-                else:
-                    p = np.array([[g.x, g.y] for g in user_staypoints.geometry])
-                labels = db.fit_predict(p)
+            # we create a new column: 'serial_num' where we assign serial numbers to each user id
+            ret_stps = ret_stps.assign(serial_num=(ret_stps["user_id"]).astype("category").cat.codes)
```

Comment:
I would suggest to change this line to `ret_stps["serial_num"] = ret_stps["user_id"].astype("category").cat.codes` for readability

Reply:
This line has been removed in the new version